vizro-core/tests/unit/vizro/actions/_callback_mapping/test_get_action_callback_mapping.py

```diff
@@ -0,0 +1,281 @@
+
+    @pytest.mark.parametrize(
+        "action_id, action_callback_outputs_expected",
+        [
+            (
+                "filter_action_filter_continent",
+                [
+                    {"component_id": "scatter_chart", "component_property": "figure"},
+                    {"component_id": "scatter_chart_2", "component_property": "figure"},
+                ],
+            ),
+            ("filter_interaction_action", [{"component_id": "scatter_chart_2", "component_property": "figure"}]),
+            (
+                "parameter_action_parameter_x",
+                [
+                    {"component_id": "scatter_chart", "component_property": "figure"},
+                    {"component_id": "scatter_chart_2", "component_property": "figure"},
+                ],
+            ),
+            (
+                "on_page_load_action_action_test_page",
+                [
+                    {"component_id": "scatter_chart", "component_property": "figure"},
+                    {"component_id": "scatter_chart_2", "component_property": "figure"},
+                ],
+            ),
+        ],
+        indirect=["action_callback_outputs_expected"],
+    )
+    def test_action_callback_mapping_outputs(self, action_id, action_callback_outputs_expected):
+        result = _get_action_callback_mapping(
+            action_id=action_id,
+            argument="outputs",
+        )
+        assert result == action_callback_outputs_expected
+
+    @pytest.mark.parametrize(
+        "export_data_action_targets, export_data_outputs_expected",
+        [
+            (None, ["scatter_chart", "scatter_chart_2"]),
+            (["scatter_chart"], ["scatter_chart"]),
+            (["scatter_chart", "scatter_chart_2"], ["scatter_chart", "scatter_chart_2"]),
+        ],
+        indirect=["export_data_outputs_expected"],
+    )
+    def test_export_data_mapping_outputs(self, export_data_action_targets, export_data_outputs_expected):
+        export_data_action = model_manager["export_data_action"]
+        export_data_action.function = (
+            export_data(targets=export_data_action_targets) if export_data_action_targets else export_data()
+        )
+
+        result = _get_action_callback_mapping(
+            action_id="export_data_action",
+            argument="outputs",
+        )
+
+        assert result == export_data_outputs_expected
+
+    @pytest.mark.parametrize(
+        "export_data_action_targets, export_data_components_expected",
+        [
+            (None, ["scatter_chart", "scatter_chart_2"]),
+            (["scatter_chart"], ["scatter_chart"]),
+            (["scatter_chart", "scatter_chart_2"], ["scatter_chart", "scatter_chart_2"]),
+        ],
+        indirect=["export_data_components_expected"],
+    )
+    def test_export_data_mapping_components(self, export_data_action_targets, export_data_components_expected):
+        export_data_action = model_manager["export_data_action"]
```

Comment:
Why can't we do this by `Action(function=...)` in the usual way instead of taking something from the model manager and then updating `action.function`?

Reply:
The reason is that we need an `action' from an existing page because targets must be calculated if they are not specified.

Reply:
Ah ok. So you're basically using the `managers_one_page_four_controls_two_graphs_filter_interaction` and then modifying the action that it provides? Would there be a nicer way to do this by parameterising `managers_one_page_four_controls_two_graphs_filter_interaction`? I can imagine this getting very complex and unreadable though, which would not be good.

If you do stick with the current strategy then it would be good to add a comment somewhere explaining why you do this because it's not obvious at a glance.

Reply:
Parametrising the fixture makes much more sense and readability isn't ruined also.